### PR TITLE
Create USACO dec20-DoYouKnowYourABCs.mdx

### PR DESCRIPTION
**Pull request by sathu-beep**
solution to https://usaco.guide/general/expected?lang=cpp 
in java, since there are a ton of solutions in CPP and the official sol is in cpp

### CODE REVIEW

File: solutions/USACO dec20-DoYouKnowYourABCs.mdx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+			if(array[i]>=largest){
+				abc = array[i];
+				largest = abc;
+				abcindex = i;
+			}
+		}
+		bc = abc - a;
+		array[aindex] = 0;
+		array[abcindex] = 0;
+		for(int i=0; i<array.length; i++){
```
Comment (dongliu0426) [inline solutions/USACO dec20-DoYouKnowYourABCs.mdx:43]:
```suggestion
		for(int i = 0; i<array.length; i++){
```

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/USACO dec20-DoYouKnowYourABCs.mdx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		array[bcindex] = 0;
+		largest = 0;
+		for(int i = 0; i<array.length; i++){
+			if(array[i]>=largest){
+				ca = array[i];
+				largest = ca;  
+			} 
+		}
+		c = ca-a;
+		b = abc-a-c;
```
Comment (maggie-j-liu) [inline solutions/USACO dec20-DoYouKnowYourABCs.mdx:57]:
```suggestion
			if(array[i] == bc){
				bcindex = i;
			}
		}
		array[bcindex] = 0;
		largest = 0;
		for(int i = 0; i < array.length; i++){
			if(array[i] >= largest){
				ca = array[i];
				largest = ca;  
			} 
		}
		c = ca - a;
		b = abc - a - c;
```

File: solutions/USACO dec20-DoYouKnowYourABCs.mdx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		int abcindex  = 0, aindex = 0, bcindex = 0;
+		int smallest = 1000000000;
+		int largest = 0;
+		for(int i = 0; i<array.length; i++){
+			if(array[i]<=smallest){
+				a = array[i];
+				smallest = a;
+				aindex = i;
+			}
+			if(array[i]>=largest){
```
Comment (maggie-j-liu) [inline solutions/USACO dec20-DoYouKnowYourABCs.mdx:34]:
```suggestion
		for(int i = 0; i < 7; i++){
			array[i] = sc.nextInt();  
		}
		int a = 0, b = 0, c = 0, abc = 0, bc, ba, ca = 0;
		int abcindex = 0, aindex = 0, bcindex = 0;
		int smallest = 1000000000;
		int largest = 0;
		for(int i = 0; i < array.length; i++){
			if(array[i] <= smallest){
				a = array[i];
				smallest = a;
				aindex = i;
			}
			if(array[i] >= largest){
```

**Review by maggie-j-liu — CHANGES_REQUESTED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]

**Review by thecodingwizard — CHANGES_REQUESTED** [review]
wait this code seems too complicated; there's a much easier solution if you just sort the array. this problem is part of the "expected knowledge" section of the Guide, so it would be better if the solution was simpler / less intimidating. Even though sorting isn't necessary for Bronze I think it's acceptable to consider that as prerequisite knowledge...

**Review by dongliu0426 — APPROVED** [review]

**Review by maggie-j-liu — APPROVED** [review]

**Pull request merged**